frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -797,10 +826,19 @@ pub mod pallet {
                 ));
             }
 
-            T::Currency::resolve_creating(&staker, reward_imbalance);
-            Self::update_staker_info(&staker, &contract_id, staker_info);
-            Self::deposit_event(Event::<T>::Reward(staker, contract_id, era, staker_reward));
+            if let Some(delegate) = RewardDelegate::<T>::get(&staker, &contract_id) {
+                // reward to the delegate
+                T::Currency::resolve_creating(&delegate, reward_imbalance);
```

Comment:
Since I changed should_restake_reward function to take into consideration if there is a delegate account, restaking will not be performed unless delegate account is not set, and in that case the reward will go to the staker.